lib/modules/manager/pipenv/artifacts.ts

```diff
@@ -120,6 +157,28 @@ export async function updateArtifacts({
         },
       ],
     };
+
+    const sourceUrl = await findPipfileSourceUrlWithCredentials(
+      newPipfileContent,
+      pipfileName
+    );
+    if (sourceUrl) {
+      logger.trace('Pipfile contains credentials');
+      const hostRule = getMatchingHostRule(sourceUrl);
+      const extraEnv = execOptions.extraEnv!;
```

Comment:
move extraEnv to a separate variable from execOptions, so er don't need the not null enforcement